debugger/memory_maps.md

```diff
@@ -0,0 +1,148 @@
+# Memory Maps
+
+The ability to understand and manipulate the memory maps of a debugged program is important for many different Reverse Engineering tasks. radare2 offers a rich set of commands to handle memory maps in the binary. This includes listing the memory maps of the currently debugged binary, removing memory maps, handling loaded libraries and more.
+
+First, let's see the help message for `dm`, the command which is responsible for handling memory maps:
+
+    [0x55f2104cf620]> dm?
+    |Usage: dm # Memory maps commands
+    | dm                               List memory maps of target process
+    | dm address size                  Allocate <size> bytes at <address> (anywhere if address is -1) in child process
+    | dm=                              List memory maps of target process (ascii-art bars)
+    | dm.                              Show map name of current address
+    | dm*                              List memmaps in radare commands
+    | dm- address                      Deallocate memory map of <address>
+    | dmd[a] [file]                    Dump current (all) debug map region to a file (from-to.dmp) (see Sd)
+    | dmh[?]                           Show map of heap
+    | dmi [addr|libname] [symname]     List symbols of target lib
+    | dmi* [addr|libname] [symname]    List symbols of target lib in radare commands
+    | dmi.                             List closest symbol to the current address
+    | dmiv                             Show address of given symbol for given lib
+    | dmj                              List memmaps in JSON format
+    | dml <file>                       Load contents of file into the current map region (see Sl)
+    | dmm[?][j*]                       List modules (libraries, binaries loaded in memory)
+    | dmp[?] <address> <size> <perms>  Change page at <address> with <size>, protection <perms> (rwx)
+    | dms[?] <id> <mapaddr>            Take memory snapshot
+    | dms- <id> <mapaddr>              Restore memory snapshot
+    | dmS [addr|libname] [sectname]    List sections of target lib
+    | dmS* [addr|libname] [sectname]   List sections of target lib in radare commands
+
+In this chapter, we'll go over some of the most useful subcommands of `dm` using simple examples. For the following examples, we'll use a simple `helloworld` program for Linux but it'll be the same for every binary.
+
+First thing first - open a program in debugging mode:
+
+    $ r2 -d helloworld
```

Comment:
explain that r2 will try to find this program in current directory and then in $PATH, even if no ./ is passed. this is contradictory with UNIX systems, but makes the behaviour consistent for windows users